test/utf32.cpp

```diff
@@ -19,24 +19,23 @@ int main() {
   // U+2603 is snowman,
   // U+20AC is the Euro sign,
   // U+2007C is a Chinese Han character that looks like three raindrops.
-  wchar_t wstr[] = L"abc\u2603\u20AC\U0002007C123 --- abc\u2603\u20AC\U0002007C123";
-  size_t wstr_len = wcslen(wstr);
+  std::wstring wstr = L"abc\u2603\u20AC\U0002007C123 --- abc\u2603\u20AC\U0002007C123";
 
   printf("sizeof(wchar_t): %d.\n", (int)sizeof(wchar_t));
 
   if (sizeof(wchar_t) == 4) {
-    utf32 *memory = malloc(wstr_len*sizeof(utf32));
```

Comment:
Is maybe problem that we are missing the null terminator here, a +4 or such?

But lgtm to land this and figure it out later.

Reply:
Sam was going to take a look when he got a chance. Though, I'm not sure I see the value in refactoring tests that use c++ features in c.

Reply:
Yeah I guess I was a little over zealous with this one, probably should have left it as C++.